Highway.cpp

```diff
@@ -13,22 +14,42 @@ void Highway::changeSpeed(int newSpeed)
 
 void Highway::addVehicleInternal(Vehicle* v)
 {
-    assert(false);
-
     /*
     depending on the derived type, call the member function that doesn't evade the cops. 
     */
+    if( auto* vCar = dynamic_cast<Car*>(v) )
+    {
+        vCar->closeWindows();
+    }
+    if( auto* vMotorcycle = dynamic_cast<Motorcycle*>(v) )
+    {
+        vMotorcycle->lanesplitAndRace();
+    }
+    if( auto* vSemiTruck = dynamic_cast<SemiTruck*>(v) )
+    {
+        vSemiTruck->closeWindows();
+    }
```

Comment:
use `else if`
You're still checking if it's a motorcycle even though it's a car.
- [ ] revise